### ISSUE TITLE
dependence predis is deprecated

### ISSUE THREAD

**Issue by jenokizm** [erikdubbelboer/phpRedisAdmin#157]
Hi, I wondered why you don’t have streams (5.0) support. I cloned the repo and thought to add it myself, but found that some methods do not work, for example
```
case 'stream':
	      echo $fullkey;
              $len = $redis->xLen($fullkey);
              break;
```

> Fatal error: Uncaught Predis\ClientException: Command 'XLEN' is not a registered Redis command.

I see composer dependence "predis/predis": "1.1.x-dev" (2017-07-12) and https://github.com/phpredis/phpredis (newer).  They abandoned the composer. Now it is distributed as a compiled library. Please replace the composer dependency in the project with the compiled one so that you can develop the project further, working with the latest redis.

**Comment by erikdubbelboer** [comment]
To be honest I don't think the users would like me switching to a compiled dependency. Right now it's super easy to setup phpRedisAdmin. With a compiled dependency it gets a lot more difficult and I'm afraid a lot of users won't be able to do this (because they don't know how or because they don't have access to install C libs).

Have you tried using this? https://github.com/nrk/predis#adding-new-commands

**Comment by jenokizm** [comment]
Thanks for the answer. I did not try these crutches. I agree with you that beginners will have difficulty installing a binary library. Of course, you can go the way that you said, but constantly adding something there is also not an ideal option. You better focus on the user-friendliness of your shell and its technical information content. For example, to debug huge databases in real time, a compiled library will do better.

**Comment by jenokizm** [comment]
> Have you tried using this? https://github.com/nrk/predis#adding-new-commands

In general, I will try this as a temporary solution. But you'd better think about changing the library.

**Comment by tealy** [comment]
I'v get an error 

Fatal error: Uncaught Predis\Response\ServerException: ERR unknown command 'SCAN' in phpRedisAdmin/vendor/src/Client.php:370 Stack trace: #0

**Comment by tealy** [comment]
> I'v get an error , my redis version was 2.2.4, trying update the redis use a new version
> 
> Fatal error: Uncaught Predis\Response\ServerException: ERR unknown command 'SCAN' in phpRedisAdmin/vendor/src/Client.php:370 Stack trace: #0

USE NEW VERSION , problem resolved

**Comment by anthosz** [comment]
Hi,

On my side, I followed the readme (git clone + git clone in vendor) and copied the config to config.inc.php (without any custom conf) to test and I have this issue:
PHP Fatal error:  Uncaught Predis\Response\ServerException: NOAUTH Authentication required. in /var/www/html/phpredisadmin/vendor/src/Client.php:354

Best regards,

**Comment by erikdubbelboer** [comment]
@anthosz its saying Redis requires a password, you have to configure it like this: https://github.com/erikdubbelboer/phpRedisAdmin/blob/1ee43ae7c2ad415cc99086575d66dfd9df6831fb/includes/config.sample.inc.php#L16

**Comment by anthosz** [comment]
> 
> 
> @anthosz its saying Redis requires a password, you have to configure it like this:
> 
> https://github.com/erikdubbelboer/phpRedisAdmin/blob/1ee43ae7c2ad415cc99086575d66dfd9df6831fb/includes/config.sample.inc.php#L16

Ok thx! It cannot be requested via form by example?

I mean, add a new server in config -> select list available on the gui -> ask to submit auth (like user/pass on phpmyadmin by example)

**Comment by erikdubbelboer** [comment]
No I'm afraid not. A pull request for this is always welcome.

**Comment by idontusenumbers** [comment]
There exists a new Predis package on Packagist. It would fix PHP 8.1 support as well.
